plugins/woocommerce-blocks/tests/e2e/tests/local-pickup/local-pickup.merchant.block_theme.spec.ts

```diff
@@ -0,0 +1,292 @@
+			);
+
+			const text = 'Enable local pickup';
+			const xpath = `//label[text()='${ text }']/preceding-sibling::span`;
+			const checkedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			await page.getByText( text ).click();
+			await utils.savelocalPickupSettings( admin, page );
+
+			const newCheckedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			expect( newCheckedState ).not.toBe( checkedState );
+		} );
+
+		test( 'Allows the title to be changed', async ( { admin, page } ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			await page
+				.locator( 'input[name="local_pickup_title"]' )
+				.fill( 'Local Pickup Test #1' );
+
+			await utils.savelocalPickupSettings( admin, page );
+
+			await expect(
+				page.locator( 'input[name="local_pickup_title"]' )
+			).toHaveValue( 'Local Pickup Test #1' );
+
+			await page
+				.locator( 'input[name="local_pickup_title"]' )
+				.fill( 'Local Pickup Test #2' );
+
+			await utils.savelocalPickupSettings( admin, page );
+
+			await expect(
+				page.locator( 'input[name="local_pickup_title"]' )
+			).toHaveValue( 'Local Pickup Test #2' );
+		} );
+
+		test( 'Allows toggling of add price field state', async ( {
+			admin,
+			page,
+		} ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			const text = 'Add a price for customers who choose local pickup';
+			const xpath = `//label[text()='${ text }']/preceding-sibling::span`;
+
+			const checkedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			await page.getByText( text ).click();
+			await utils.savelocalPickupSettings( admin, page );
+
+			const newCheckedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			expect( newCheckedState ).not.toBe( checkedState );
+		} );
+
+		test( 'Cost and tax status can be set', async ( { admin, page } ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			const text = 'Add a price for customers who choose local pickup';
+			const input = page.locator(
+				`//label[text()='${ text }']/preceding-sibling::span`
+			);
+
+			// eslint-disable-next-line playwright/no-conditional-in-test
+			if ( ! ( await input.locator( 'svg' ).isVisible() ) ) {
+				await page.getByText( text ).click();
+			}
+
+			await page
+				.locator( 'input[name="local_pickup_cost"]' )
+				.fill( '20' );
+
+			await page.getByLabel( 'Taxes' ).selectOption( 'none' );
+
+			await utils.savelocalPickupSettings( admin, page );
+
+			await expect(
+				page.locator( 'input[name="local_pickup_cost"]' )
+			).toHaveValue( '20' );
+
+			await expect(
+				page.locator( 'select[name="local_pickup_tax_status"]' )
+			).toHaveValue( 'none' );
```

Comment:
What are your thoughts on checking both scenarios `Taxable` and `Non taxable` and potentially verifying the changes in the frontend?